dkc/core/rest/folder.py

```diff
@@ -27,11 +33,15 @@ class FolderViewSet(ModelViewSet):
         AllowAny,
         # IsAuthenticatedOrReadOnly,
     ]
-    serializer_class = FolderSerializer
 
     filter_backends = [ActionSpecificFilterBackend]
     filterset_class = FoldersFilterSet
 
+    def get_serializer_class(self):
+        if self.request.method in ['PUT', 'PATCH']:
```

Comment:
```suggestion
        if self.action in ['update', 'partial_update']:
```

This is a more direct coupling to how DRF actually dispatches to the action methods which are the handlers.